core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1888,9 +1890,12 @@ private long loadFileMetadataAndJournal(LockedInodePath inodePath,
             .setRecursive(options.isCreateAncestors()).setMetadataLoad(true).setPersisted(true);
     String ufsOwner = ufs.getOwner(ufsUri.toString());
     String ufsGroup = ufs.getGroup(ufsUri.toString());
-    short ufsPermission = ufs.getMode(ufsUri.toString());
-    createFileOptions = createFileOptions.setPermission(
-        new Permission(ufsOwner, ufsGroup, ufsPermission));
+    short ufsMode = ufs.getMode(ufsUri.toString());
+    Permission permission = new Permission(ufsOwner, ufsGroup, ufsMode);
+    if (resolution.getShared()) {
+      permission.getMode().applyExtraOtherBits(permission.getMode().getOwnerBits());
```

Comment:
Provide `set*Bits` instead of `applyExtra*Bits` and do the following:

```
Mode mode = permission.getMode();
mode.setOtherBits(mode.getOtherBits().or(mode.getOwnerBits));
```

IMHO, this is more readable and unlike `applyExtra*Bits`, `set*Bits` can be use modularly with other Bits-level operations such as `and`, and `imply`.